### ISSUE TITLE
save needs appropriate sorting of paths into datasets as well

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#1140]
I guess, since I have created a new file in a few datasets and wanted to save that with one command but didn't work since save didn't care:
```shell
$> datalad save ds000017A/dataset_description.json
[WARNING] Was instructed to commit ['/mnt/btrfs/datasets-dev-20161125/datalad/crawl/openfmri/ds000017A/dataset_description.json'] files but repository is not dirty 
```
hard to say about which repository it was talking about since both superdataset and that ds000017A are dirty:
```
$> datalad ls . ds000017A
.   [git]  master  - 2016-11-25/16:42:04  X
ds000017A   [annex]  master  1.1.0+1-4-ge778b1d 2016-11-25/15:07:36  X

$> cd ds000017A 
README                    demographics.txt  models/              study_key.txt  sub003/  sub006/  task_key.txt
changelog.txt             license.txt       release_history.txt  sub001/        sub004/  sub007/
dataset_description.json  model_key.txt*    scan_key.txt         sub002/        sub005/  sub008/
(venv-tests)3 11148.....................................:Fri 25 Nov 2016 04:44:24 PM EST:.
(git)smaug:\u2026nt/btrfs/datasets-dev-20161125/datalad/crawl/openfmri/ds000017A[master]git
$> git status
On branch master
Your branch is ahead of 'datalad-public-dev/master' by 1 commit.
  (use "git push" to publish your local commits)
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	dataset_description.json
```

datalad as of 167903253287a735c48874dcde206698771a4769

**Comment by mih** [comment]
FTR: I started with this and discovered several issues that become apparent once inputs are sorted upfront. This will be a larger RF.

**Comment by mih** [comment]
Sorting implemented in #1169

**Comment by mih** [comment]
Fix merged.
